airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -441,3 +444,10 @@ def monitor_launched_pod(self, launcher, pod) -> Tuple[State, Optional[str]]:
                 'Pod returned a failure: {state}'.format(state=final_state)
             )
         return final_state, result
+
+    def on_kill(self) -> None:
+        if self.pod:
+            pod: k8s.V1Pod = self.pod
+            namespace = pod.metadata.namespace
+            name = pod.metadata.name
+            self.client.delete_namespaced_pod(name=name, namespace=namespace, grace_period_seconds=0)
```

Comment:
Isn't gracePeriodSeconds something else than how long before the pod is stopped -- this controls how long the pod is kept around for in the API server, but not when the pod is terminated, isn't it?

(I remember we investigated this for an EKS fix around short lived pods and increasing this didn't help.)

Reply:
@ashb grace_period_seconds is a recommendation, raising it doesn't necessarily mean the ApiServer will wait that long. By setting it to 0 we are telling the ApiServer "Delete this before doing anything else" rather than "get around to it eventually"